.travis.yml

```diff
@@ -1,30 +1,82 @@
-dist: trusty
 sudo: required
-language: python
+language: generic
+
 cache:
   directories:
     - $HOME/.ccache
 notifications:
   slack:
     on_success: never
     on_failure : never
-python:
-  - "2.7"
-  - "3.4"
-  - "3.5"
-  - "3.6"
-before_install:
-  - export PATH="/usr/lib/ccache:$PATH"
-  - ccache -M 256M && ccache -s
-  - sudo apt-get install libgmp-dev libmpfr-dev libmpfi-dev libmpc-dev cython texlive texlive-latex-extra latexmk
-  # note: sphinx in ubuntu is version 1.2.2 that does not support
-  # sphinx.ext.imgmath. We just install the last version.
-  - pip install sphinx
+
+matrix:
+  include:
+    - os: linux
+      dist: trusty
+      language: python
+      python: 2.7
+
+    - os: linux
+      dist: trusty
+      language: python
+      python: 3.5
+
+    - os: linux
+      dist: trusty
+      language: python
+      python: 3.6
+
+    - os: osx
+      env: PYTHON_VERSION=2.7
+      osx_image: xcode6.4
+
+    - os: osx
+      env: PYTHON_VERSION=3.5
+      osx_image: xcode6.4
+
+    - os: osx
+      env: PYTHON_VERSION=3.6
+      osx_image: xcode6.4
+
 install:
-  - pip install --verbose .
+  - |
+    if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+      echo ""
+      echo "Installing a fresh version of Miniconda."
+      MINICONDA_URL="https://repo.continuum.io/miniconda"
+      MINICONDA_FILE="Miniconda3-latest-MacOSX-x86_64.sh"
+      curl -L -O "${MINICONDA_URL}/${MINICONDA_FILE}"
+      bash $MINICONDA_FILE -b
+      # Configure conda.
+      echo ""
+      echo "Configuring conda."
+      source $HOME/miniconda3/bin/activate root
+      conda config --add channels conda-forge
+      conda config --set show_channel_urls true
+      conda config --set always_yes yes --set changeps1 no
+      echo ""
+      echo "Creating conda environment"
+      conda create -n test python=${PYTHON_VERSION} gmp mpfr mpc
+      source activate test
+      echo ""
+      echo "Installing latex"
+      port install texlive texlive-latex-extra
+      python setup.py install --shared=$CONDA_PREFIX;
+    else # Linux
+      export PATH="/usr/lib/ccache:$PATH"
+      ccache -M 256M && ccache -s
+      sudo apt-get install libgmp-dev libmpfr-dev libmpfi-dev libmpc-dev cython texlive texlive-latex-extra latexmk
+      # note: sphinx in ubuntu is version 1.2.2 that does not support
+      # sphinx.ext.imgmath. We just install the last version.
+      pip install --verbose .
+    fi
+
 script:
+  - pip install sphinx
```

Comment:
this should not be here. `pip install sphinx` is part of the installation process.

Reply:
right